pkg/storage/trb_advice_letter.go

```diff
@@ -199,3 +199,44 @@ func (s *Store) GetTRBAdviceLetterByTRBRequestID(logger *zap.Logger, trbRequestI
 
 	return &letter, nil
 }
+
+// GetTRBAdviceLetterStatusByTRBRequestID fetches a TRB advice letter's status, if present, by its associated request's ID.
+// It does _not_ return an error if no advice letter is found; this is an expected scenario if an advice letter hasn't been created,
+// in which case resolver logic will compute the advice letter's status.
+func (s *Store) GetTRBAdviceLetterStatusByTRBRequestID(logger *zap.Logger, trbRequestID uuid.UUID) (*models.TRBAdviceLetterStatus, error) {
```

Comment:
The problem is that `GetTRBAdviceLetterByTRBRequestID()` logs _any_ error returned from the query (see line 180) before doing the `errors.Is(err, sql.ErrNoRows)` check. That log message is what triggers the alert.

That said, thinking about it, I could just rewrite `GetTRBAdviceLetterByTRBRequestID()` slightly and have it do the check for `sql.ErrNoRows` before logging anything. Then I could either write `GetTRBAdviceLetterStatusByTRBRequestID()` as a wrapper, like you suggest, or push the logic of checking for a `ResourceNotFoundError` back up to the resolver, like it was before this PR. It would mean that we wouldn't get an alert if something else is calling `GetTRBAdviceLetterByTRBRequestID()` with a wrong ID, though.